src/main/java/org/webcurator/core/targets/MembersRemovedEventPropagator.java

```diff
@@ -151,6 +151,16 @@ private void recurseDown(AbstractTarget aTarget) {
 	 * @param propagate True to propagate scheduling; otherwise false.
 	 */
 	private void recurseUp(TargetGroup ancestor, boolean propagate) {
+		Map<Object,Object> duplicateValidator=new HashMap<>();
```

Comment:
I think recurseUpInternal should be called here

Reply:
Exactly. Thanks a lot Hanna.